java/flight/flight-sql/src/test/java/org/apache/arrow/flight/sql/example/FlightSqlStatelessExample.java

```diff
@@ -0,0 +1,246 @@
+                    .bindAll().build();
+
+            while (binder.next()) {
+              executeQuery(statement, listener);
+            }
+          }
+        }
+      } catch (StreamCorruptedException e) {
+        // Case where there are no parameters
+        final String query = new String(command.getPreparedStatementHandle().toString("UTF-8"));
+        final Connection connection = dataSource.getConnection();
+        final PreparedStatement preparedStatement = connection.prepareStatement(query,
+                ResultSet.TYPE_SCROLL_INSENSITIVE, ResultSet.CONCUR_READ_ONLY);
+        executeQuery(preparedStatement, listener);
+      }
+    } catch (final SQLException | IOException | ClassNotFoundException e) {
+      LOGGER.error(format("Failed to getStreamPreparedStatement: <%s>.", e.getMessage()), e);
+      listener.error(CallStatus.INTERNAL.withDescription("Failed to prepare statement: " + e).toRuntimeException());
+    } finally {
+      listener.completed();
+    }
+  }
+
+  private void executeQuery(PreparedStatement statement,
+                            final ServerStreamListener listener) throws IOException, SQLException {
+    try (final ResultSet resultSet = statement.executeQuery()) {
+      final Schema schema = jdbcToArrowSchema(resultSet.getMetaData(), DEFAULT_CALENDAR);
+      try (final VectorSchemaRoot vectorSchemaRoot = VectorSchemaRoot.create(schema, rootAllocator)) {
+        final VectorLoader loader = new VectorLoader(vectorSchemaRoot);
+        listener.start(vectorSchemaRoot);
+
+        final ArrowVectorIterator iterator = sqlToArrowVectorIterator(resultSet, rootAllocator);
+        while (iterator.hasNext()) {
+          final VectorSchemaRoot batch = iterator.next();
+          if (batch.getRowCount() == 0) {
+            break;
+          }
+          final VectorUnloader unloader = new VectorUnloader(batch);
+          loader.load(unloader.getRecordBatch());
+          listener.putNext();
+          vectorSchemaRoot.clear();
+        }
+        listener.putNext();
+      }
+    }
+  }
+
+  @Override
+  public FlightInfo getFlightInfoPreparedStatement(final CommandPreparedStatementQuery command,
+                                                   final CallContext context,
+                                                   final FlightDescriptor descriptor) {
+    String query = null;
+    try {
+      final byte[] handle = command.getPreparedStatementHandle().toByteArray();
+      try (ByteArrayInputStream bis = new ByteArrayInputStream(handle);
+           ObjectInputStream ois = new ObjectInputStream(bis)) {
+        final DoPutPreparedStatementResultPOJO doPutPreparedStatementResultPOJO =
+                (DoPutPreparedStatementResultPOJO) ois.readObject();
+        query = doPutPreparedStatementResultPOJO.getQuery();
+      } catch (StreamCorruptedException e) {
+        query = new String(command.getPreparedStatementHandle().toString("UTF-8"));
```

Comment:
Nit: use Charsets.UTF_8 instead of the "UTF-8" string.

Reply:
Sorry, forgot to push a commit. I had changed this to
command.getPreparedStatementHandle().toStringUtf8()
as this seemed most appropriate.